src/streaming-client.ts

```diff
@@ -145,20 +153,21 @@ export class RevAiStreamingClient extends EventEmitter {
                     }
                 }
             });
-            this.doSendLoop(connection, this.requests);
+            this.doSendLoop(conn, this.requests);
         });
     }
 
-    private doSendLoop(connection: any, buffer: PassThrough): void {
-        if (connection.connected) {
+    private doSendLoop(conn: connection, buffer: PassThrough): void {
+        if (conn.connected) {
             let value = buffer.read(buffer.readableLength);
             if (value !== null) {
-                connection.send(value);
-                if (value.includes('EOS') || value.includes(Buffer.from('EOS'))) {
-                    connection.sendUTF('EOS');
+                if (Buffer.compare(value, Buffer.from('EOS')) === 0) {
+                    conn.sendUTF('EOS');
+                } else {
+                    conn.send(value);
```

Comment:
If read buffer is just EOS, then we'll send EOS. If not, we send the full buffer.

Reply:
This was the state of the sdk some time ago. We need to be careful here because we're reading buffer.readableLength. this means if there are multiple messages written to the buffer well read all of them at once. This why we had to check for contains(EOS) and not just equals(EOS). Does enabling writableObjectMode somehow change the way buffer.readableLength works so we're guaranteed to only get one chunk at a time?

Reply:
@kbridbur I actually enabled `objectMode` to the wrong stream - I needed to enable it in the underlying `requests` stream - here's the fix commit - https://github.com/revdotcom/revai-node-sdk/pull/76/commits/c3c9b0593808d0ef05eccecf0d50f8d11716da82 - now when reading, if EOS was provided as a string, it will be read as a string, therefore we can check typing and content when reading.

To answer your question about reading and object mode. If in object mode, read will always return a single object no matter the `readableLength` (which in this case is the number of objects now instead of number of bytes) - [source](https://nodejs.org/api/stream.html#:~:text=always%20return%20a%20single%20item)